### DISCUSSION TITLE
Would an AStarGrid2D implementation for hexagonal grids be worth it?

### DISCUSSION THREAD

**Discussion by PsychonicJoe** [godotengine/godot-proposals#8291] [2D]
I'm super new so I wouldn't understand how to do it now but recently a need for it has popped up in one of my project ideas and I was thinking if I should invest the time into learning and implementing it.
Or should I just work on a local implementation in my own project.

The current AStarGrid2D works fine to some extent on the hexagonal grid but since each cell has 6 neighboring cells, there's always 2 disconnected cells causing the pathfinder to go to one of the other connected neighboring cells first then back to the original target which is redundant and looks very janky.

I don't know how the current implementation works exactly but I'm reading around but this is all new to me so I'd like to hear what you guys think.

**Comment by m21-cerutti** [comment]
Related to https://github.com/godotengine/godot-proposals/issues/9710
